### PR TITLE
New bid adapter: Wipes

### PR DESCRIPTION
**Pull request by Tosh39**
## Type of change
- [x] New bidder adapter   

## Description of change

- test parameters for validating bids
```
{
  bidder: 'wipes',
  params: {
    asid: 'XXXXXXXXXX'
  }
}
```

- contact email of the adapter’s maintainer

contact@3-shake.com

- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- Bidder Params Documentation PR: https://github.com/prebid/prebid.github.io/pull/1894

### CODE REVIEW

File: modules/wipesBidAdapter.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# Description
+
+Connect to WIPES for bids.
+Publishers needs to be set up and approved by WIPES team to enable this adapter.
+Please contact support@wipestream.com for further information.
+
+# Test Parameters
+```javascript
+var adUnits = [
```
Comment (musikele) [inline modules/wipesBidAdapter.md:17]: Would you mind specifying also the media type configuration ? I mean mediaType.context - instream, outstream, etc. 

This is not mandatory but it is appreciated to test the bidder.

**Comment by Tosh39** [comment]
@musikele 
Thank you for checking my code.

I updated test parameter to make test easier.

**Comment by musikele** [comment]
Hello @Tosh39 , 
I still see a couple of issues. 
For example, you adjusted the example data this way: 

```
bids: [{
         bidder: 'wipes',
         params: {
           asid: 'dWyPondh2EGB_bNlrVjzIXRZO9F0k1dpo0I8ZvQ',
           mediaTypes: {
              video: {
                 context: 'outstream'
              }
           },
         }
       }]
```

but in my tests, to get bids, I had to do this: 
```
mediaTypes: {
                video: {
                    context: 'outstream'
                }
            },
            bids: [{
                bidder: 'wipes',
                params: {
                    asid: 'dWyPondh2EGB_bNlrVjzIXRZO9F0k1dpo0I8ZvQ',
                }
            }]
```

Anyway, even if I get bids, I then see this error: 

```
ERROR: Invalid bid from wipes. Ignoring bid: Video bid does not have required vastUrl or renderer property
```

I tried to add a renderer to the ad unit, without success. 

Am I missing something? 

(Also - the response requestId is different from the request. I know this is only a fake response but I had to tweak prebid code in order to validate it.)

**Comment by Tosh39** [comment]
@musikele 
Thank you for reviewing my code. I misunderstood about mediaType and our tool doesn't return video info, only html tag, so I changed mediaType to banner from video.
It's corrected now.

Could you check again please?

File: modules/wipesBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      width: response.width,
+      height: response.height,
+      creativeId: response.videoCreativeId || 0,
+      dealId: response.dealId,
+      currency: 'JPY',
+      netRevenue: netRevenue,
+      ttl: config.getConfig('_bidderTimeout'),
+      referrer: bidRequest.data.r || '',
+      mediaType: BANNER,
+      ad: response.adTag,
```
Comment (musikele) [inline modules/wipesBidAdapter.js:59]: In the network response I see `ad_tag`, not `adTag` -- please fix it 

Also, check all other camelCase variables: netRevenue, videoCreativeId, dealId

Comment (Tosh39) [reply modules/wipesBidAdapter.js]: @musikele 
I'm sorry, I updated again. It's ready to be reviewed

File: modules/wipesBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const params = bidRequest.params;
+    const asid = params.asid;
+    return {
+      method: 'GET',
+      url: ENDPOINT_URL,
+      data: {
+        asid: asid,
+        mediaTypes: {
+          video: {
+            context: 'outstream'
```
Comment (musikele) [inline modules/wipesBidAdapter.js:33]: is the whole `mediaTypes` part here is not needed anymore?

**Comment by Tosh39** [comment]
@musikele 
I guess that's because of network problem, so could you move forward please?

And I am sorry about adapter code. It's fixed.

**Pull request merged**